P4RT_App_DB_Schema.md

```diff
@@ -0,0 +1,396 @@
+| <name>              =   | String / Integer | ;  The value format comes from the action type.                                                                                                                                                                                                                                                                    |
+| meter/cir            =  | <integer>        | ; Committed Information Rate. Packets meeting this rate limit are treated as GREEN. All other packets are marked RED or YELLOW (see meter/pir).                                                                                                                                                                    |
+| meter/cburst       =    | <integer>        | ; Committed Burst Rate. [Required if meter/cir is present.] Packets meeting this rate limit are treated as GREEN.                                                                                                                                                                                                  |
+| meter/pir             = | <integer>        | ; Peak Information Rate. Packets meeting this rate limit and not meter/cir, meter/cburst are treated as YELLOW. Packets not meeting this rate limit are treated as RED. For RED/GREEN-only processing, meter/pir & meter/pburst can be omitted or set to the same values as meter/cir & meter/cburst respectively. |
+| meter/pburst       =    | <integer>        | ; Peak Burst Rate. [Required if meter/pir is present.] See meter/pir for more information.                                                                                                                                                                                                                         |
+
+```
+127.0.0.1:6379> hgetall "P4RT:ACL_ACL_INGRESS_TABLE:{\"match/dst_mac\":\"33:33:00:00:00:02&ff:ff:ff:ff:ff:ff\",\"match/icmpv6_type\":\"0x87&0xff\",\"match/ip_protocol\":\"0x3a&0xff\",\"match/is_ipv6\":\"0x1\",\"priority\":2070}"
+ 1) "action"
+ 2) "trap"
+ 3) "param/qos_queue"
+ 4) "0x6"
+ 5) "meter/cir"
+ 6) "28000"
+ 7) "meter/cburst"
+ 8) "7000"
+ 9) "meter/pir"
+10) "28000"
+11) "meter/pburst"
+12) "7000"
+13) "controller_metadata"
+14) "my metadata"
+
+127.0.0.1:6379> hgetall "P4RT:ACL_ACL_PRE_INGRESS_TABLE:{\"match/dst_ip\":\"10.53.192.0&255.255.240.0\",\"match/is_ipv4\":\"0x1\",\"priority\":1132}"
+1) "param/vrf_id"
+2) "p4rt-vrf-80"
+3) "action"
+4) "set_vrf"
+5) "controller_metadata"
+6) "my metadata"
+
+
+```
+
+
+
+##### 3. FIXED_* Tables
+
+FIXED_* Tables are similar to ACL_* Tables. However, FIXED_* Tables have a predefined set of matches & actions. This prevents the need for a DEFINITION table for these tables.
```

Comment:
for most of the FIXED table, there are already existing orch handling the those tables, for example, the neighorch handles the neighbor table, the routeroch handles the route table. with this design, in the orchagent, we now have two orch handling the same sai objects. It appears that current design introduce redundancy to handle same sai objects, and is causing architecture divergence in orchagent. It also means P4RT orch could not leverage the investment of sonic community on existing orchagent. what is the reason to have separate table instead of reusing existing app db, like ROUTE_TABLE, NEXTHOP_TABLE, and etc. on? What is the north star here?

Reply:
The existing orchagent code does not meet some of the requirements of the remote controller path such as response path, application ownership tags etc. Multiple writers(local and remote applications) to the same table without tags would cause additional issues. The new path can be merged to the existing path after these functionalities are built in the existing path. The north star is a single path in the future which meets the requirements for local and remote paths.